source/guide_wsgidav.rst

```diff
@@ -0,0 +1,200 @@
+
+.. tag:: web
+.. tag:: file-storage
+.. tag:: webdav
+
+.. sidebar:: About
+
+  .. image:: _static/images/wsgidav.png
+      :align: center
+
+############
+WsgiDAV
+############
+
+.. tag_list::
+
+
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`supervisord <daemons-supervisord>`
+  * :manual:`domains <web-domains>`
+
+License
+=======
+
+MIT license.
+
+  * https://github.com/mar10/wsgidav/blob/master/LICENSE
+
+
+Installation
+============
+
+Use pip3 to install wsgidav:
+
+::
+
+ [isabell@stardust ~]$ pip3 install wsgidav cheroot --user
+
+Create the data folder:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/webdav
+
+
+Add additional folders you want to access via webdav with:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/webdav/<newfoldername>
+
+config file
+======================
+
+.. note::
+
+    Only use spaces to indent in the config file! No tabs!
+
+
+The following template is the minimum to get wsgidav up and running. You have to change provider_mapping, user_mapping and the port (if another service is already running on 8080).
+If you miss some functionality or want more information, take a look at the sample file and the explanation of the configuration options:
+https://wsgidav.readthedocs.io/en/latest/user_guide_configure.html#sample-wsgidav-yaml
+
+
+Config file path:
+
+::
+
+ ~/.wsgidav/wsgidav.yaml
+ 
+
+
+Config file:
+
+::
+
+ server: "cheroot"
+ server_args:
+     numthreads: 256
+ host: 0.0.0.0
+ port: 8080
+ block_size: 8192
+ add_header_MS_Author_Via: true
+ hotfixes:
+     winxp_accept_root_share_login: false
+     win_accept_anonymous_options: false
+     unquote_path_info: false
+     re_encode_path_info: null
+     emulate_win32_lastmod: true
+ 
+ middleware_stack:
+     - wsgidav.debug_filter.WsgiDavDebugFilter
+     - wsgidav.error_printer.ErrorPrinter
+     - wsgidav.http_authenticator.HTTPAuthenticator
+     - wsgidav.dir_browser.WsgiDavDirBrowser
+     - wsgidav.request_resolver.RequestResolver
+ 
+ 
+ mount_path: null
+ provider_mapping:
+     "/": "/home/isabell/webdav/"
+     "/firstpath": "/home/isabell/webdav/<newpath>"
+ # Adds folder "/home/isabell/webdav/<newpath>" to URL https://isabell.uber.space/firstpath
+	
+ simple_dc:
+     user_mapping:
+         "*":  # default (used for all shares that are not explicitly listed). Keep this one!
+             "isabell":
+                 password: "<password>" # Change it!
+                 roles: ["editor"]
+         "/firstpath": # The mount_path this user has access to.
+             "isabellscousin": # Username
+                 password: "<password>" # Password. Change it!
+ # Here you can add additional users (same indent)
+
+ http_authenticator:
+     accept_basic: true
+     accept_digest: true
+     default_to_digest: true
+     trusted_auth_header: null
+     domain_controller: null
+ 
+ nt_dc:
+     preset_domain: null
+     preset_server: null
+ pam_dc:
```

Comment:
Enabling PAM will allow any unrelated user on the system to login using their SSH username/password. Please disable it.